src/go/accessd/config/config.go

```diff
@@ -0,0 +1,165 @@
+	if strings.HasPrefix(target, ipv4Scheme+":") {
+		return resolver.Target{
+			Scheme:   ipv4Scheme,
+			Endpoint: target[len(ipv4Scheme)+1:],
+		}
+	}
+	if strings.HasPrefix(target, ipv6Scheme+":") {
+		return resolver.Target{
+			Scheme:   ipv6Scheme,
+			Endpoint: target[len(ipv6Scheme)+1:],
+		}
+	}
+	return grpcutil.ParseTarget(target, false)
+}
+
+// Configer returns a parsed config.
+type Configer interface {
+	Config() *mconfig.AccessD
+}
+
+// ConfigManager implements Configer via a loaded config.
+type ConfigManager struct {
+	config *mconfig.AccessD
+}
+
+var defaultConfig = &mconfig.AccessD{
+	LogLevel:                        mconfig.AccessD_INFO,
+	SctpdDownstreamServiceTarget:    "unix:///tmp/sctpd_downstream.sock",
+	SctpdUpstreamServiceTarget:      "unix:///tmp/sctpd_upstream.sock",
+	MmeSctpdDownstreamServiceTarget: "unix:///tmp/mme_sctpd_downstream.sock",
+	MmeSctpdUpstreamServiceTarget:   "unix:///tmp/mme_sctpd_upstream.sock",
+}
+
+const (
+	cStyleCommentStart = "/*"
+	cStyleCommentEnd   = "*/"
+)
+
+func filterCStyleComments(in string) string {
+	var filtered string
+	for {
+		idx := strings.Index(in, cStyleCommentStart)
+		if idx == -1 {
+			filtered += in
+			break
+		}
+		filtered += in[:idx]
+		in = in[idx:]
+		idx = strings.Index(in, cStyleCommentEnd)
+		if idx == -1 {
+			break
+		}
+		in = in[idx+2:]
+	}
+	return filtered
+}
+
+func filterJSONComments(in string) string {
+	var filtered []string
+	for _, line := range strings.Split(filterCStyleComments(in), "\n") {
+		if idx := strings.Index(line, "//"); idx != -1 {
+			line = line[0:idx]
+		}
+		line = strings.TrimSpace(line)
+		if line != "" {
+			filtered = append(filtered, line)
+		}
+	}
+	return strings.Join(filtered, "\n")
+}
+
+func loadMConfigProto(
+	readFile func(string) ([]byte, error),
+	unmarshalProto func([]byte, proto.Message) error,
+	path string,
+) (*mconfig.AccessD, error) {
+	bytes, err := readFile(path)
+	if err != nil {
+		return nil, errors.Wrap(err, "path="+path)
+	}
+	filtered := []byte(filterJSONComments(string(bytes)))
+	config := &mconfig.AccessD{}
+	if err := unmarshalProto(filtered, config); err != nil {
+		return nil, errors.Wrapf(
+			err,
+			"path=%s filtered=%s",
+			path,
+			string(filtered))
+	}
+	return config, nil
+}
+
+// NewConfigManager constructs a *ConfigManager from the provided file path.
+func NewConfigManager(configFilePath string) (*ConfigManager, error) {
+	config, ok := proto.Clone(defaultConfig).(*mconfig.AccessD)
+	if !ok {
+		panic("clone of defaultConfig not *mconfig.AccessD")
+	}
+	if _, err := os.Stat(configFilePath); err == nil {
+		loaded, err := loadMConfigProto(
+			ioutil.ReadFile, protojson.Unmarshal, configFilePath)
+		if err != nil {
+			return nil, err
+		}
+		proto.Merge(config, loaded)
+	}
+	return &ConfigManager{config: config}, nil
+}
+
+// Config returns the parsed config contained in ConfigManager.
+func (c *ConfigManager) Config() *mconfig.AccessD {
+	return c.config
+}
```

Comment:
this maybe a naive question, but it looks like you are loading mconfig file

Why can't we use `GetServiceConfig` we use in the other gateways [here](https://sourcegraph.com/-/editor?remote_url=git%40github.com%3Auri200%2Fmagma.git&branch=state_service_add_print_grpc&file=feg%2Fgateway%2Fservices%2Fs8_proxy%2Fservicers%2Fconfig.go&editor=JetBrains&version=v1.2.1&utm_product_name=IntelliJ+IDEA&utm_product_version=2021.1.3&start_row=39&start_col=29&end_row=39&end_col=29). 



Note that our mcofigs values come in string format and sometimes they need some massaging like [here](https://sourcegraph.com/-/editor?remote_url=git%40github.com%3Auri200%2Fmagma.git&branch=state_service_add_print_grpc&file=feg%2Fgateway%2Fservices%2Fs8_proxy%2Fservicers%2Fconfig.go&editor=JetBrains&version=v1.2.1&utm_product_name=IntelliJ+IDEA&utm_product_version=2021.1.3&start_row=47&start_col=31&end_row=47&end_col=31)

Reply:
trying to see if there's a easier way to do configs. we will follow up here in a sprint or two as part of #8832 with a potentially bigger redesign of config (also with a path towards unifying a new config strategy with config across the rest of our services/codebase)